modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutionServiceImpl.java

```diff
@@ -336,13 +336,19 @@ private CompletableFuture<AsyncDataCursor<InternalSqlRow>> executeQuery(
 
         CompletableFuture<Void> firstPageReady0 = firstPageReady;
 
-        Predicate<String> nodeExclusionFilter = operationContext.nodeExclusionFilter(); 
+        Predicate<String> nodeExclusionFilter = operationContext.nodeExclusionFilter();
 
-        return queryManager.execute(tx, plan, nodeExclusionFilter).thenApply(dataCursor -> new TxAwareAsyncCursor<>(
+        CompletableFuture<AsyncDataCursor<InternalSqlRow>> f = queryManager.execute(tx, plan,
```

Comment:
let's add the test to `ExecutionServiceImplTest` which covers this particular case (you need to make sure that transaction is rolled back if distributed query were unable to initialize)